packages/shared/components/inputs/RecipientInput.svelte

```diff
@@ -19,28 +22,44 @@
         recipient?.type === 'account'
             ? { key: recipient.account.name, value: recipient.account.depositAddress }
             : { value: recipient?.address }
-    const accountOptions: IOption[] = $visibleActiveAccounts
-        ?.filter((account) => account.index !== $selectedAccountIndex)
-        ?.map((account) => ({
-            id: account.index,
-            key: account.name,
-            value: account.depositAddress,
-        }))
 
+    $: accountOptions = isLayer2 ? <IOption[]>[] : getLayer1AccountOptions()
     $: recipient = getSubjectFromAddress(selected?.value)
+    $: isLayer2, (error = '')
 
     export function validate(): Promise<void> {
         if (recipient?.type === 'address') {
             if (!recipient.address) {
-                error = 'Recipient is required'
-                return Promise.reject(error)
+                error = localize('error.send.recipientRequired')
             }
-            error = validateBech32Address(addressPrefix, recipient?.address)
-            if (error) {
-                return Promise.reject(error)
+            if (isLayer2) {
+                error = validateEthereumAddress(recipient?.address)
+            } else {
+                error = validateBech32Address(addressPrefix, recipient?.address)
             }
+        } else if (recipient?.type === 'account') {
+            if (isLayer2) {
+                error = localize('error.layer2.layer1Recipient')
+            }
+        } else {
+            error = localize('error.send.recipientRequired')
+        }
+
+        if (error) {
+            return Promise.reject(error)
+        } else {
+            return Promise.resolve()
         }
-        return Promise.resolve()
+    }
+
+    function getLayer1AccountOptions(): IOption[] {
+        return $visibleActiveAccounts
+            ?.filter((account) => account.index !== $selectedAccountIndex)
+            ?.map((account) => ({
```

Comment:
are the `?` required here?

Reply:
Not really, although we have it in a lot of places in the code. I removed them :relaxed: